docs/source/integrations/ibis.md

```diff
@@ -0,0 +1,407 @@
+            data.execute().to_sql(self._table_name, conn.con, if_exists="replace")
+
+    def _describe(self) -> Dict[str, Any]:
+        """Return a description of the dataset."""
+        return {
+            "table_name": self._table_name,
+            "has_query": self._query is not None
+        }
+```
+
+## Configuring the DataCatalog
+
+Now you can configure your DataCatalog to use the Ibis dataset. Add the following to your `conf/base/catalog.yml`:
+
+```yaml
+# conf/base/catalog.yml
+customers_table:
+  type: <package_name>.extras.datasets.ibis_dataset.IbisTableDataSet
+  connection: ${ibis_connection}
+  table_name: customers
+
+filtered_customers:
+  type: <package_name>.extras.datasets.ibis_dataset.IbisTableDataSet
+  connection: ${ibis_connection}
+  query: ${ibis_query}
+```
+
+You'll need to provide the connection and query objects in your `catalog_dict` when creating the `DataCatalog`:
+
+```python
+from kedro.framework.context import KedroContext
+from kedro.framework.hooks import hook_impl
+
+
+class ProjectContext(KedroContext):
+    @hook_impl
+    def register_catalog(self, catalog, credentials, load_versions, save_version, journal):
+        # Get the Ibis connection from the context
+        ibis_connection = lambda: self.ibis_conn
+        
+        # Define an example query
+        def ibis_query():
+            customers = self.ibis_conn.table('customers')
+            return customers.filter(customers.age > 30)
+        
+        # Add these objects to the catalog's dictionary
+        catalog.add_feed_dict({
+            'ibis_connection': ibis_connection,
+            'ibis_query': ibis_query()
+        })
+        
+        return catalog
+```
+
+## Using Ibis in Kedro nodes
+
+Now you can use Ibis in your Kedro nodes to perform SQL operations. Here's an example of a node that uses Ibis to transform data:
+
+```python
+def filter_customers(customers_table: ibis.expr.types.TableExpr) -> ibis.expr.types.TableExpr:
+    """Filter customers by age and calculate average purchase amount."""
+    # Filter customers older than 25
+    filtered = customers_table.filter(customers_table.age > 25)
+    
+    # Group by customer_id and calculate average purchase amount
+    result = filtered.group_by('customer_id').aggregate(
+        avg_purchase=filtered.purchase_amount.mean(),
+        total_purchases=filtered.purchase_amount.count()
+    )
+    
+    return result
+
+
+def join_customer_orders(customers: ibis.expr.types.TableExpr, 
+                        orders: ibis.expr.types.TableExpr) -> ibis.expr.types.TableExpr:
+    """Join customers with their orders."""
+    # Join customers and orders on customer_id
+    joined = customers.join(orders, customers.id == orders.customer_id)
+    
+    # Select relevant columns
+    result = joined.select([
+        customers.id,
+        customers.name,
+        orders.order_date,
+        orders.amount
+    ])
+    
+    return result
```

Comment:
These look mostly good to me 👍🏼 Could you maybe pass `ruff` over them? So that double quotes are used instead of single quotes etc